security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -58,14 +62,25 @@ type sdsservice struct {
 // Assert we implement the generator interface
 var _ model.XdsResourceGenerator = &sdsservice{}
 
+// NewXdsServer builds a minimal DiscoveryServer for istio-agent SDS only.
+// The DiscoveryServer was originally designed for Istiod, so a lot of this is plugging in fake empty information.
 func NewXdsServer(stop chan struct{}, gen model.XdsResourceGenerator) *xds.DiscoveryServer {
-	s := xds.NewXDS(stop)
+	// Setup a mostly empty NewEnvironment
+	env := model.NewEnvironment()
+	env.ServiceDiscovery = aggregate.NewController(aggregate.Options{})
+	env.ConfigStore = memory.Make(collection.NewSchemasBuilder().Build())
+	env.Watcher = meshcfg.NewFixedWatcher(meshcfg.DefaultMeshConfig())
+	env.PushContext().Mesh = env.Watcher.Mesh()
+	env.Init()
+
+	s := xds.NewDiscoveryServer(env, "istiod", "", map[string]string{})
```

Comment:
nit: This is the node agent, so why istiod not leave it to empty

Reply:
this is unused entirely, I will clean up. And cluster ID as well